recordedit/form.controller.js

```diff
@@ -425,15 +425,36 @@
             return submissionRow;
         }
 
-        function removeFormRow(index) {
+        function spliceRows(index) {
             vm.recordEditModel.rows.splice(index, 1);
+            vm.recordEditModel.oldRows.splice(index, 1);
             vm.recordEditModel.submissionRows.splice(index, 1);
+            vm.recordEditModel.foreignKeyData.splice(index, 1);
+            if (vm.editMode) $rootScope.tuples.splice(index, 1);
             $timeout(function() {
                 onResize();
                 $rootScope.showSpinner = false;
             }, 10);
         }
 
+        var CONFIRM_DELETE =  (chaiseConfig.confirmDelete === undefined || chaiseConfig.confirmDelete) ? true : false;
```

Comment:
This is not really important, but since we refactored the `chaiseConfig`, you don't need to check for it anymore and you should be able to use the `chaiseConfig.confirmDelete` directly.